dispatcher/MongoConnect.py

```diff
@@ -294,12 +300,17 @@ def is_timeout(self, doc, t):
         ret = False
         if dt > self.timeout:
             self.log.debug(f'{host} last reported {int(dt)} sec ago')
-            ret |= True
+            ret = ret or True
         if has_ackd is not None and t - has_ackd > self.timeout_take_action:
             self.log.debug(f'{host} hasn\'t ackd a command from {int(t-has_ackd)} sec ago')
             if self.host_config[host] == 'tpc':
-                self.hypervisor.handle_timeout(host)
-            ret |= True
+                dt = (now() - self.hv_timeout_fix[host]).total_seconds()
```

Comment:
Do we want the possibility of resetting the hosts asynchronous or would you want to reset the `self.hv_timeout_fix[host]` for all the hosts simultaneously? 

Since a host timeing out should not per se affect the other host, I think your current approach makes sense

Reply:
I've thought about this. Resetting a process takes ~30-40s or so, which is very long compared to the 3s logic cycle in the dispatcher. If we have multiple hosts timing out, resetting them in series (current implementation) is very slow and the system appears to stop responding while this is ongoing. On the other hand, if we have host resetting happen asynchronously, we don't want the dispatcher running control logic while hosts are being reset. We could modify the list of logical detectors for that loop if a reset is in progress to bypass this, but this is tricky to implement.

Reply:
The timeout is currently implemented per-host because they can crash independently. We want to avoid not fixing a problem on reader1 just because we just fixed an issue on reader2.